### DISCUSSION TITLE
ORM configuration check -- is there a method for that?

### DISCUSSION THREAD

**Discussion by taigit** [sqlalchemy/sqlalchemy#9904] [Usage Questions]
I'm debugging my ORM configuration by instantiating a row.
I'm wondering if there is a method I can call that will explicitly check the ORM configuration.
It would be nice as a development-time operation that is separate from runtime.

I assume that MetaData.create_all() does some checking, but there seems to be lots of other checking after that too.

Django admin has the check command:
https://docs.djangoproject.com/en/4.2/ref/django-admin/#check
just wondering if  sqlalchemy has something similar.

**Comment by zzzeek** [comment]
we dont have something that works "after the fact" at that point, SQLAlchemy tags configurational issues as they arise, either by emitting warnings or errors.   If your setup has no warnings or errors, then SQLAlchemy has not detected any known configurational problems.

**Reply by zzzeek** [reply]
if you have specific examples of something that went wrong, feel free to share.
